pkg/operator/starter.go

```diff
@@ -185,3 +191,31 @@ func replaceNamespaceFunc(namespace string) resourceapply.AssetFunc {
 		return bytes.Replace(content, []byte(namespaceReplaceKey), []byte(namespace), -1), nil
 	}
 }
+
+func generateCredentialsRequest(namespace string) resourceapply.AssetFunc {
+	return func(name string) ([]byte, error) {
+		content, err := assets.ReadFile(name)
+		if err != nil {
+			panic(err)
+		}
+
+		stsRoleARN := os.Getenv(stsIAMRoleARNEnvVar)
+		cloudTokenPath := ""
+		if stsRoleARN != "" {
+			// The cluster is in STS mode. Add double quotes around string values, just to be sure
+			stsRoleARN = "\"" + stsRoleARN + "\""
+			cloudTokenPath = "\"" + cloudTokenPathValue + "\""
+		} else {
+			// Non-STS mode. Set the values explicitly to the empty value ""
+			stsRoleARN = "\"\""
+			cloudTokenPath = "\"\""
+		}
+		klog.V(4).Infof("Using ARN %s and token path %s", stsRoleARN, cloudTokenPath)
+
+		content = bytes.Replace(content, []byte(namespaceReplaceKey), []byte(namespace), -1)
+		content = bytes.Replace(content, []byte(stsIAMRoleARNKey), []byte(stsRoleARN), -1)
+		content = bytes.Replace(content, []byte(cloudTokenPathKey), []byte(cloudTokenPath), -1)
+
+		return content, nil
+	}
+}
```

Comment:
LGTM, but I'd personally vote for moving this logic to a manifest hook. Since we'd need to add hook support in the controller, we could take the chance and move the NS setting logic to a hook as well, which would make the end result cleaner IMO.

But that's just my opinion, which is not strongly held. I'd be fine with this approach as well.

Reply:
I added hooks to CredentialsRequestController, passing more `Unstructured` around.